### PR TITLE
[libpng] Update to 1.6.44, add feature tools

### PR DESCRIPTION
**Pull request by Osyotr**
https://github.com/pnggroup/libpng/compare/v1.6.43...v1.6.44

Note: `libpng` now installs `PNGConfig.cmake`. CMake wrapper is removed. `libpng-config.cmake` is kept for backwards compatibility.

### CODE REVIEW

**Comment by dg0yt** [comment]
Could `libpng-config.cmake` be replaced with a forwarding+warning script like `ports/xmlsec/xmlsec-config.cmake`?

File: ports/libpng/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -1,3 +0,0 @@
-find_library(PNG_LIBRARY_RELEASE NAMES png16 libpng16 NAMES_PER_DIR PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/lib" NO_DEFAULT_PATH)
-find_library(PNG_LIBRARY_DEBUG NAMES png16d libpng16d NAMES_PER_DIR PATHS "${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug/lib" NO_DEFAULT_PATH)
-_find_package(${ARGS})
```
Comment (dg0yt) [inline ports/libpng/vcpkg-cmake-wrapper.cmake:3]: This cannot be removed. It is required for `find_package(PNG)`, using CMake's Find module.

File: ports/libpng/vcpkg.json [hunk 2]
```diff
@@ -21,9 +20,19 @@
     },
     "zlib"
   ],
+  "default-features": [
+    {
+      "name": "tools",
+      "platform": "!android & !ios & !static"
+    }
+  ],
```
Comment (dg0yt) [inline ports/libpng/vcpkg.json:28]: To be removed before merge.

File: ports/libpng/vcpkg.json [hunk 3]
```diff
@@ -21,9 +20,19 @@
...
   "features": {
     "apng": {
       "description": "This is backward compatible with the regular libpng, both in library usage and format"
+    },
+    "tools": {
+      "description": "Build the libpng tools",
+      "supports": "!android & !ios & !static"
```
Comment (dg0yt) [inline ports/libpng/vcpkg.json:35]: IMO The package must be patched to build the tools with the chosen library linkage.

File: ports/libpng/vcpkg-cmake-wrapper.cmake (hunk 1 above)
Comment (Osyotr) [reply ports/libpng/vcpkg-cmake-wrapper.cmake:3]: I've noticed that the file is installed to `file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/png")`
Shouldn't it be installed in "${CURRENT_PACKAGES_DIR}/share/PNG" instead?

Comment (Osyotr) [reply ports/libpng/vcpkg-cmake-wrapper.cmake:3]: Also, can I just force `CONFIG` and remove `find_library` calls?

Comment (dg0yt) [reply ports/libpng/vcpkg-cmake-wrapper.cmake]: Lower-case dir name.

Forcing CONFIG may work when the config provides everything which the find modules provides (mostly variables). Libxml2 does that.

Until you hit a project which uses a custom find module. (Such as GDAL for SQLite3...)

Comment (dg0yt) [reply ports/libpng/vcpkg-cmake-wrapper.cmake]: Basically, there is no need to remove working wrappers.

File: ports/libpng/portfile.cmake [hunk 4]
```diff
@@ -77,18 +85,22 @@ vcpkg_cmake_configure(
...
         -DSKIP_INSTALL_EXECUTABLES=ON
         -DSKIP_INSTALL_FILES=OFF
+        ${FEATURE_OPTIONS}
     OPTIONS_DEBUG
         -DSKIP_INSTALL_HEADERS=ON
     MAYBE_UNUSED_VARIABLES
         PNG_ARM_NEON
 )
 vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(PACKAGE_NAME PNG CONFIG_PATH lib/cmake/PNG)
```
Comment (dg0yt) [inline ports/libpng/portfile.cmake:97]:
```suggestion
vcpkg_cmake_config_fixup(PACKAGE_NAME png CONFIG_PATH lib/cmake/PNG)
```
It is just used for the target dir name, lower case is preferred for share in vcpkg, and cmake accepts lower case.

File: ports/libpng/libpng-config.cmake [hunk 5]
```diff
@@ -0,0 +1,12 @@
+file(READ "${CMAKE_CURRENT_LIST_DIR}/usage" usage)
+message(WARNING "find_package(libpng) is deprecated.\n${usage}")
+
+include(CMakeFindDependencyMacro)
+find_dependency(PNG CONFIG REQUIRED)
```
Comment (WangWeiLin-MV) [inline ports/libpng/libpng-config.cmake:5]:
```suggestion
find_dependency(PNG CONFIG)
```

File: ports/libpng/usage [hunk 6]
```diff
@@ -1,4 +1,4 @@
-The package libpng is compatible with built-in CMake targets:
+libpng provides CMake targets:
 
-    find_package(PNG REQUIRED)
-    target_link_libraries(main PRIVATE PNG::PNG)
+  find_package(PNG REQUIRED)
```
Comment (WangWeiLin-MV) [inline ports/libpng/usage:3]:
```suggestion
  find_package(PNG CONFIG REQUIRED)
```

Comment (dg0yt) [reply ports/libpng/usage:3]: IMO this should be left as is. The new config is available with the libpng version, but the new libpng version isn't yet available everywhere.

File: ports/libpng/libpng-config.cmake (hunk 5 above)
Comment (Osyotr) [reply ports/libpng/libpng-config.cmake:5]: There's no difference. Not worth resetting CI over this.

Comment (Osyotr) [reply ports/libpng/libpng-config.cmake:5]: >  But others will copy the bad example, and they won't believe that there is a difference.

Well...
https://github.com/microsoft/vcpkg/commit/d5dd03b545b08a5df1ab11361cbb014a1da50cd6#diff-d40fdf2b5523a10b3ee5de1b598ff8bc5491770a18df22520c7b46653caccbafR4

**Review by WangWeiLin-MV — DISMISSED** [review]
The port features usage tests pass with the following triplets:

* x64-windows

File: ports/libpng/vcpkg.json [hunk 7]
```diff
@@ -21,9 +20,19 @@
     },
     "zlib"
   ],
+  "default-features": [
+    {
+      "name": "tools",
+      "platform": "!android & !ios"
+    }
+  ],
```
Comment (vicroms) [inline ports/libpng/vcpkg.json:28]: I don't think `tools` should be enabled by default.

https://learn.microsoft.com/vcpkg/contributing/maintainer-guide#default-features-should-enable-behaviors-not-apis

**Review by vicroms — CHANGES_REQUESTED** [review]

**Review by vicroms — APPROVED** [review]

**Pull request merged**